src/main/java/de/tum/in/www1/artemis/service/ExamService.java

```diff
@@ -539,9 +540,30 @@ public void validateForStudentExamGeneration(Exam exam) throws BadRequestAlertEx
         List<StudentDTO> notFoundStudentsDtos = new ArrayList<>();
         for (var studentDto : studentDtos) {
             var registrationNumber = studentDto.getRegistrationNumber();
+
+            // If no registration number is given try to fetch it via the login
+            if (!StringUtils.hasText(registrationNumber)) {
+                var userOptional = userService.getUserByLogin(studentDto.getLogin());
+                if (userOptional.isEmpty()) {
+                    notFoundStudentsDtos.add(studentDto);
+                    continue;
+                }
+
+                registrationNumber = userOptional.get().getRegistrationNumber();
+            }
```

Comment:
I think this part will very likely introduce unneeded database calls. May I suggest to (a) either only check login as a fallback or (b) update the `findUserWithGroupsAndAuthoritiesByRegistrationNumber` to search for either/or?

Reply:
This is already a fallback `a` if I'm not mistaken? We only call this if there is no registration number!

Reply:
I'll push an update reducing the worst case amount of database calls to `1` in a bit.